selectable_pdf_infra/selectable_pdf_infra/selectable_pdf_stack.py

```diff
@@ -169,74 +180,91 @@ def __init__(
-        ]
-        for policy in managed_policies:
-            process_textract_lambda.role.add_managed_policy(
-                aws_iam.ManagedPolicy.from_aws_managed_policy_name(policy)
+        process_textract_lambda.role.add_managed_policy(
+                aws_iam.ManagedPolicy.from_aws_managed_policy_name('AmazonTextractFullAccess')
+            )
+        process_textract_lambda.role.attach_inline_policy(
+            self.get_policy_write_to_ddb_table(ddb_documents_table)
+        )
+        process_textract_lambda.role.attach_inline_policy(
+            aws_iam.Policy(self, 'S3ReadWriteObject',
+                statements=[aws_iam.PolicyStatement(actions=['s3:GetObject','s3:PutObject'],
+                    resources=[doc_bucket.bucket_arn+'/*',processed_bucket.bucket_arn+'/*'])]
             )
+        )
+        process_textract_lambda.role.attach_inline_policy(
+            aws_iam.Policy(self, 'SqsPublishMessage',
+                statements=[aws_iam.PolicyStatement(actions=['sqs:SendMessage'],
+                    resources=[processed_textracted_queue_sqs.queue_arn])]
+            )
+        )
         # set the trigger
         textract_job_topic.add_subscription(
             aws_sns_subscriptions.LambdaSubscription(process_textract_lambda)
         )
 
         # Lambda function turning a scanned PDF (i.e. a PDF where we cannot select 
         # text) into a searchable PDF (i.e. a PDF where we can select text). This function 
-        # recieved message from sqs, therefore its timeout MUST be smaller than the 
+        # received message from sqs, therefore its timeout MUST be smaller than the 
         # visibility timeout of the source SQS, otherwise, cyclic call!!.
         lambda_timeout_sec = sqs_visibility_timeout_sec-1 #second
-        searchable_pdf_lambda = aws_lambda.Function(
+        selectable_pdf_lambda = aws_lambda.Function(
             self,
             id='SelectablePdf',
-            runtime=aws_lambda.Runtime.JAVA_11,
-            handler='LambdaSqs::handleRequest',
-            code=aws_lambda.Code.asset(self.build_selectable_pdf_lib()),
-            timeout=core.Duration.seconds(lambda_timeout_sec),
+            runtime=aws_lambda.Runtime.PYTHON_3_8,
+            handler='main.lambda_handler',
+            code=aws_lambda.Code.from_asset(os.path.join(LAMBDA_DIRPATH, 'selectable_pdf')),
+            layers=[textracttools_layer, helpertools_layer],
+            timeout=Duration.seconds(lambda_timeout_sec),
             environment={
+                'DDB_DOCUMENTS_TABLE': ddb_documents_table.table_name,
                 'OUTPUT_BUCKET': processed_bucket.bucket_name,
+                'LOG_LEVEL': 'INFO',
+                'ADD_WORD_BBOX': '0',
+                'SHOW_CHARACTER': '0',
+                'PDF_IMAGE_DPI': '200',
+
             },
             retry_attempts=0,
-            memory_size=4048,  # 4GB to get a big CPU with the Lambda. Code is CPU intensive
+            memory_size=3000
```

Comment:
I think he even 2GB would suffice, have you done some testing here?

Reply:
yes, 2GB works.